File: platform/node/test/fixtures/canned-agent.mjs
/**
 * Test fixture: a canned agent built on the SDK runtime, no model call.
 * Streams a fixed reply so the runtime loop can be verified deterministically.
 */
import { run } from "../../src/runtime.js";

run({
  name: "canned-agent",
  async onPrompt(ctx) {
    ctx.assistantStart();
    ctx.reasoningDelta("deciding");
    for (const w of ["hello", " from", " canned", " agent"]) {
      ctx.assistantDelta(w);
    }
    ctx.assistantEnd("hello from canned agent");
  },
});

File: platform/python/tests/fixtures/canned_agent.py
"""Test fixture: a canned Python agent built on the SDK runtime, no model call."""

from llmtoolforge_agent import run


def on_prompt(ctx):
    ctx.assistant_start()
    ctx.reasoning_delta("deciding")
    for word in ["hello", " from", " canned", " agent"]:
        ctx.assistant_delta(word)
    ctx.assistant_end("hello from canned agent")


if __name__ == "__main__":
    run(on_prompt, name="canned-agent")
